### PR TITLE
feat: delete and duplicate selection

### PR DESCRIPTION
**Pull request by AraxTheCoder**
This adds the ability to delete and duplicate the selection as requested in #170.
These two new options get shown in the toolOptions collapsible menu, as seen in the screenshot.

A screenshot of Saber on an Android device with a new select bar at the bottom with a duplicate button and delete button

I still need to copy the "Delete" from imageOptions of all languages to selectionOptions, but when I tried it so far, other translations got replaced with the missing English translation.
Any help on how I can do this correctly is welcome.

### CODE REVIEW

File: lib/components/toolbar/toolbar.dart [hunk 1]
```diff
@@ -330,8 +341,17 @@ class _ToolbarState extends State<Toolbar> {
...
                 enabled: !widget.readOnly,
                 onPressed: () {
-                  toolOptionsType.value = ToolOptions.hide;
-                  widget.setTool(Select.currentSelect);
+                  if (widget.currentTool == Select.currentSelect) {
+                    if (toolOptionsType.value == ToolOptions.select) {
+                      toolOptionsType.value = ToolOptions.hide;
+                    } else {
+                      toolOptionsType.value = ToolOptions.select;
+                    }
```
Comment (adil192) [inline lib/components/toolbar/toolbar.dart:349]: Please can you make this a conditional operator:
```dart
toolOptionsType.value = toolOptionsType.value == ToolOptions.select
    ? ToolOptions.hide
    : ToolOptions.select;
```

File: lib/components/toolbar/toolbar.dart [hunk 2]
```diff
@@ -330,8 +341,17 @@ class _ToolbarState extends State<Toolbar> {
...
+                  } else {
+                    // Default showing selection functions
```
Comment (adil192) [inline lib/components/toolbar/toolbar.dart:351]:
```suggestion
                    // Show the selection bar by default
```

File: lib/pages/editor/editor.dart [hunk 3]
```diff
@@ -1245,6 +1245,61 @@ class EditorState extends State<Editor> {
             });
           },
           currentTool: currentTool,
+          duplicateSelection: () {
+            setState(() {
+              final select = currentTool as Select;
+              if(select.doneSelecting){
```
Comment (adil192) [inline lib/pages/editor/editor.dart:1251]: Prefer early returns in situations like this
```dart
if (select.doneSelecting) return;
```

**Review by adil192 — CHANGES_REQUESTED** [review]

**Comment by adil192** [comment]
> I still need to copy the "Delete" from imageOptions of all languages to selectionOptions, but when I tried it so far, other translations got replaced with the missing English translation. Any help on how I can do this correctly is welcome.

I typically do this as a separate commit/PR to keep the diff readable.
That being said, were you using a command like
```bash
dart run slang edit copy path.to.imageOptions.delete path.to.selectionOptions.delete
```

File: lib/i18n/strings.i18n.yaml [hunk 4]
```diff
@@ -251,6 +251,9 @@ editor:
     setAsBackground: Set as background
     removeAsBackground: Remove as background
     delete: Delete
+  selectionOptions:
```
Comment (adil192) [inline lib/i18n/strings.i18n.yaml:254]: Please can you also change this to `selectionBar:`

**Comment by AraxTheCoder** [comment]
> > I still need to copy the "Delete" from imageOptions of all languages to selectionOptions, but when I tried it so far, other translations got replaced with the missing English translation. Any help on how I can do this correctly is welcome.
> 
> I typically do this as a separate commit/PR to keep the diff readable. That being said, were you using a command like
> 
> ```shell
> dart run slang edit copy path.to.imageOptions.delete path.to.selectionOptions.delete
> ```

Ok so you take care of this yourself?

**Review by adil192 — APPROVED** [review]

**Comment by adil192** [comment]
Thank you!

**Pull request merged**